protocol/superchainERC20/convert.md

```diff
@@ -0,0 +1,176 @@
+
+We prefer simplicity and will therefore focus on the Isolated approach, i.e., a single `SuperchainERC20` per legacy token. For a more detailed discussion, please refer to the Appendix.
+
+# Proposed Solution: Isolated Convert
+
+## Implementation
+
+The Convert Method will require modifying the `L2StandardBridge` to include the following:
+
+- `convertToSuper(address _*legacyAddr, address _superAddr, uint256 _amount)`: converts* `_amount` *legacy tokens with address `_*legacyAddr` to the same `_amount` of `SuperchainERC20` tokens in address `_superAddr`.
+- `convertFromSuper(address _*legacyAddr, address superAddr, uint256 _amount)`: converts* `_amount` *`SuperchainERC20` with address `_super*Addr` to the same `_amount` of legacy tokens in address `_legacyAddr`.
+
+These functions will:
+
+1. Check that the input and output addresses are valid.
+2. `burn` the input address balance from the caller.
+3. `mint` the output address balance to the caller.
+
+An example implementation that depends on the `isValidPair()` access control function looks like this:
+
+```solidity
+function convertToSuper(address _legacyAddr, address _superAddr, uint256 _amount) public {
+  require(isValidPair(_legacyAddr, _superAddr), "Invalid address pair");
+
+  IERC20(_legacyAddr).burn(msg.sender, _amount);
+  IERC20(_superAddr).mint(msg.sender, _amount);
+}
+
+function convertFromSuper(address _legacyAddr, address _superAddr, uint256 _amount) public {
+  require(isValidPair(_legacyAddr, _superAddr), "Invalid address pair");
+
+  IERC20(_superAddr).burn(msg.sender, _amount);
+  IERC20(_legacyAddr).mint(msg.sender, _amount);
+}
+```
+
+The corresponding `SuperchainERC20` will need to allow the `L2StandardBridge` to mint and burn tokens. We will discuss the `isValidPair()` function in the [section below](#access-control).
+
+> 💡 Note:
+> We considered placing the `convert()` function directly in the `SuperchainERC20`, similar to how WETH manages deposits and withdrawals. However, we believe including them in the `L2StandardBridge` makes more sense. This is because:
+>
+> - It makes the token leaner. Users will interact with `convert()` only a few times and in a context similar to deposit and withdraw.
+> - The `SuperchainERC20` representation for migrated liquidity will be as close as possible to newly deployed `SuperchainERC20`s.
+> - The `L2StandardBridge` needs to be involved because of burn and mint. Having the functionality in the token would have added additional external calls.
+
+### Access Control
+
+The contract will need to:
+
+- **Check `OptimismMintableERC20` address is valid**: there is no need to verify anything as any liquidity risk will be isolated.
+- **Check `SuperchainERC20` is valid**: There are two ways to implement access control in the Isolated design.
+  - **Registry**: If the Factory stores the `SuperchainERC20` address for each legacy address.
```

Comment:
How does the factory choose to store an L2 super-erc20 contract address value by a `_legacyAddr` L2-address key? Does it store it upon deployment of the erc20? What happens to previously deployed addresses?